homeassistant/components/pglab/__init__.py

```diff
@@ -0,0 +1,107 @@
+"""PG LAB Electronics integration."""
+
+from __future__ import annotations
+
+from pypglab.mqtt import Client, Sub_State, Subcribe_CallBack
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.components.mqtt.subscription import (
+    async_prepare_subscribe_topics,
+    async_subscribe_topics,
+    async_unsubscribe_topics,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, async_get_hass
+
+from .const import DEVICE_ALREADY_DISCOVERED, DISCONNECT_COMPONENT, DOMAIN
+from .discovery import create_discovery
+
+# The discovery instance
+DISCOVERY_INSTANCE = "pglab_discovery_instance"
+
+# Supported platforms
+PLATFORMS = [
+    Platform.SWITCH,
+]
+
+
+async def mqtt_publish_callback(
+    topic: str, payload: str, qos: int, retain: bool
+) -> None:
+    """Define the call back for pglab module to publish a mqtt message."""
+    hass = async_get_hass()
```

Comment:
Don't use `async_get_hass` if not needed.
Either make this again an inner function (I didn't notice it had the purpose of giving access to the `hass` object), or pass partial functions to the `Client` constructor.

Same comment for subscribe and unsubscribe.

Reply:
Ok ... restoring as before.